src/server/svr_func.c

```diff
@@ -5102,6 +5103,28 @@ check_and_run_jobs(struct prov_vnode_info * prov_vnode_info)
 
 	rc = is_runnable(pjob, prov_vnode_info);
 	if (rc == 0) {
+		if (do_sync_mom_hookfiles || sync_mom_hookfiles_proc_running) {
+
+			/**
+			 * If mom hook files sync is in process then create
+			 * a time task where you perform this check again,
+			 * and start the job once it is done
+			 */
+
+			DBPRT(("%s: setting the time task as sync mom"
+				"hookfiles is not completed\n"
+				, __func__))
+
+			/* set a work task to run after 5 sec from now */ 
+			if (!set_task(WORK_Timed, time_now + 5, 
+				is_hooks_sync_done, pjob)){
+				log_err(errno, __func__,
+					"Unable to set task for is_hooks_sync_done; requeuing the job");
+				(void)force_reque(pjob);
+			}
+			return;
```

Comment:
I'm confused with this change. Can you explain a little more.
check_and_run_jobs is called from is_vnode_prov_done function. Before this change check_and_run_jobs was definitive in the sense that it use to either run the job or mark it as failed. Now this will make job wait until hooks are synced. But the caller of this function will go ahead and delete the provisioning record from server's tracking table while it is still in progress. The caller may even start provisioning for other nodes which means that we may end up provisioning more than the max concurrent provisioning limit.